### PR TITLE
[EPT-1715] create page modal v2

### PR DESCRIPTION
**Pull request by shanthru**
[EPT-1715](https://sailthru.atlassian.net/browse/EPT-1715)

Adds a modal to create a new HTML page

Please checkout PHP https://github.com/sailthru/php/pull/5229 as a mock backend for creating pages.

[EPT-1715]: https://sailthru.atlassian.net/browse/EPT-1715?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[EPT-1715]: https://sailthru.atlassian.net/browse/EPT-1715?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    const alphanumericRegex = /^\d*[a-zA-Z0-9][a-zA-Z0-9]*$/;
+    const allowDashInName = name.replace(new RegExp("-", "g"), "");
+    if (name.length < 1) {
+      setNamingError("Please enter a page name.");
+      return;
+    }
+    if (!allowDashInName.match(alphanumericRegex)) {
+      setNamingError("Page names must be alphanumeric.");
+      return;
+    }
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:32]: thoughts / suggestions:
- (non-blocking) The variables `allowDashInName` and `alphanumericRegex` initially confused me because of their location at the top of the method. I thought they were needed for the entire method because of this placement. I'd suggest moving that logic closer to where it is used (right above the if-statement)
- (blocking / question) The variable `allowDashInName` name confused because I was expecting it to be a boolean value when its actually a new name variable. I think you are using it to check for alphanumeric characters after dashes have been removed. I'd suggest that you write one regular expression that checks for all valid characters (see suggestion below). Is there are problem with this approach? Are dashes only allowed in certain places?
- (blocking) The error message should match the explainer text from the ticket (see suggest below).

```suggestion
    if (name.length < 1) {
      setNamingError("Please enter a page name.");
      return;
    }
    
    const allowedCharactersRegex = /^[a-zA-Z0-9-]+$/; // only alphanumeric characters and dashes allowed
    if (!name.match(allowedCharactersRegex)) {
      setNamingError("Page names can only contain alphanumeric characters and dashes.");
      return;
    }
```

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    if (name === "oc") {
+      setNamingError(
+        "The page name 'oc' is a reserved page name. Please select a new page name."
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:35]: nit-pick
```suggestion
        "The name 'oc' is a reserved page name. Please use a different page name."
```

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  onChange,
+  isDuplicateName,
+  isNameChanged,
+}) {
+  const [namingError, setNamingError] = useState(false);
+  const [optoutNamingError, setOptoutNamingError] = useState(false);
+
+  // function to check for empty, alphanumeric, and 'oc' page names
+  const handleNamingError = function () {
+    setNamingError("");
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:22]: question: Should `setOptoutNamingError` also be reset here?

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      return;
+    }
+    if (name == "optout") {
+      setOptoutNamingError(
+        "The page name 'optout' requires 'User Management' as the category. Please select 'User Management' below."
+      );
+      return;
+    }
+    setNamingError(false);
+    setOptoutNamingError(false);
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:46]: question: In the beginning of this method, the parameter used for `setNamingError` is an empty string. Why is a boolean value being used here?
```suggestion
    setNamingError("");
    setOptoutNamingError("");
```

Comment (lindsaycooper) [reply src/ListView/features/CreatePage/CreatePageForm.jsx:46]: question: If this is being done at the top of the method, why does it need to be done again here?

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      setNamingError("Page names must be alphanumeric.");
+      return;
+    }
+    if (name === "oc") {
+      setNamingError(
+        "The page name 'oc' is a reserved page name. Please select a new page name."
+      );
+      return;
+    }
+    if (name == "optout") {
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:39]: nit-pick: I think this should be a `===`
```suggestion
    if (name === "optout") {
```

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+      setOptoutNamingError(
+        "The page name 'optout' requires 'User Management' as the category. Please select 'User Management' below."
+      );
+      return;
+    }
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:44]: nit-pick (non-blocking): These simpler checks could be moved higher up in the method. I'd suggest the order: length check, reserved name check, regex check

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    setNamingError(false);
+    setOptoutNamingError(false);
+    return;
+  };
+
+  // sets the error if the name is a duplicate (returned from the backend)
+  useEffect(() => {
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:51]: nit-pick: I think your code explains this well enough without the comment (good variable names below 👍 )
```suggestion
  useEffect(() => {
```

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    if (isDuplicateName) {
+      setNamingError(isDuplicateName);
+    }
+  }, [isDuplicateName]);
+
+  // function to reset the name error when the name is changed
+  const onChangeResetNamingError = (e) => {
+    if (isNameChanged) {
+      setNamingError(false);
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:60]: question (same as above): Should this be an empty string instead?

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+  // sets the error if the name is a duplicate (returned from the backend)
+  useEffect(() => {
+    if (isDuplicateName) {
+      setNamingError(isDuplicateName);
+    }
+  }, [isDuplicateName]);
+
+  // function to reset the name error when the name is changed
+  const onChangeResetNamingError = (e) => {
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:58]: nit-pick: I think your code explains this well enough without the comment (good function name 👍 )
```suggestion
  const onChangeResetNamingError = (e) => {
```

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+  name = "",
+  onChange,
+  isDuplicateName,
+  isNameChanged,
+}) {
+  const [namingError, setNamingError] = useState(false);
+  const [optoutNamingError, setOptoutNamingError] = useState(false);
+
+  // function to check for empty, alphanumeric, and 'oc' page names
+  const handleNamingError = function () {
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:21]: nit-pick: I think your code explains this well enough without the comment
```suggestion
  const handleNamingError = function () {
```

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+  return (
+    <Fieldset>
+      <FormGroup>
+        <Label error={namingError}>Name</Label>
+        <Input
+          autoFocus
+          value={name}
+          onChange={onChangeResetNamingError}
+          onBlur={handleNamingError}
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx:73]: nit-pick (non-blocking): 
```suggestion
          onBlur={determineNamingError}
```

**Review by lindsaycooper — COMMENTED** [review]
@shanthru This might make more sense to chat in person about:
I think the directory structure here is a bit confusing.
- What belongs in the `feature` subdirectory?
- What belongs in `core`?
- What belongs in `src` (no sub-directory)?

File: src/ListView/features/CreatePage/CreatePageForm.jsx (hunk 2 above)
Comment (shanthru) [reply src/ListView/features/CreatePage/CreatePageForm.jsx]: updated

File: src/ListView/features/CreatePage/CreatePageModal.jsx [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+  function handleCreate() {
+    pagesApi
+      .createPage({ ...state, mode })
+      .then(function (response) {
+        if (response._data.editor_url) {
+          window.location.href = `${window.location.origin}${response._data.editor_url}`;
+        } else {
+          // this should never happen
+          setError("Something went wrong and we could not load the page.");
+          console.log("An error occurred redirecting to the new page");
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageModal.jsx]: suggestion: Add more helpful debugging information here:
```suggestion
          console.error(`Error redirecting to page origin:${window.location.origin}, url:${response._data.editor_url}`);
          
```

File: src/ListView/features/CreatePage/CreatePageModal.jsx [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+          window.location.href = `${window.location.origin}/hosted-pages-list`;
+        }
+        return response._data;
+      })
+      .catch(function (serverError) {
+        let duplicateNameError = "There is already a page named";
+        let serverErrorMessage = serverError.error_message;
+        if (
+          serverErrorMessage &&
+          serverErrorMessage.includes(duplicateNameError)
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageModal.jsx]: nit-pick: I don't think the variable name helps with clarity here
```suggestion
        let serverErrorMessage = serverError.error_message;
        if (
          serverErrorMessage &&
          serverErrorMessage.includes("There is already a page named")
```

File: src/ListView/features/CreatePage/CreatePageModal.jsx [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+        ) {
+          setIsDuplicateName("Unique page name required.");
+        } else {
+          console.log("ERROR: ", serverError.error_message);
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageModal.jsx]:
```suggestion
          console.error("ERROR: ", serverErrorMessage);
```

File: src/ListView/features/CreatePage/CreatePageModal.jsx [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+        return response._data;
+      })
+      .catch(function (serverError) {
+        let duplicateNameError = "There is already a page named";
+        let serverErrorMessage = serverError.error_message;
+        if (
+          serverErrorMessage &&
+          serverErrorMessage.includes(duplicateNameError)
+        ) {
+          setIsDuplicateName("Unique page name required.");
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageModal.jsx]: suggestion: Elsewhere in this file `setIsDuplicateName` uses booleans instead of strings. I think this makes sense. This error message can be moved to where it's displayed.
```suggestion
          setIsDuplicateName(true);
```

File: src/ListView/features/CreatePage/CreatePageForm.jsx [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+        "Page names can only contain alphanumeric characters and dashes."
+      );
+      return;
+    }
+  };
+
+  // sets the error if the name is a duplicate (returned from the backend)
+  useEffect(() => {
+    if (isDuplicateName) {
+      setNamingError(isDuplicateName);
```
Comment (lindsaycooper) [inline src/ListView/features/CreatePage/CreatePageForm.jsx]:
```suggestion
      setNamingError("Unique page name required.");
```

**Review by lindsaycooper — CHANGES_REQUESTED** [review]

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+      setOptoutNamingError(
+        "The page name 'optout' requires 'User Management' as the category. Please select 'User Management' below."
+      );
+      return;
+    }
+    const alphanumericRegex = /^[a-zA-Z0-9]+$/;
+    const nameWithoutDashes = name.replace(new RegExp("-", "g"), "");
+    if (!nameWithoutDashes.match(alphanumericRegex)) {
+      setNamingError(
+        "Page names can only contain alphanumeric characters and dashes."
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:44]: nit-pick: Trying to better match the other error message:
```suggestion
        "The page name must only contain alphanumeric characters and dashes. Please use a different page name."
```

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+      );
+      return;
+    }
+  };
+
+  // sets the error if the name is a duplicate (returned from the backend)
+  useEffect(() => {
+    if (isDuplicateName) {
+      setNamingError(isDuplicateName);
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:53]:
```suggestion
      setNamingError("The page name must be unique. Please use a different page name.");
```

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+        <Label error={namingError}>Name</Label>
+        <Subtext>
+          Page name must follow URL standards and only contain alphanumeric
+          characters and dashes. It is recommended to use dashes instead of
+          spaces.
+        </Subtext>
+        <Input
+          autoFocus
+          value={name}
+          onChange={onChangeResetNamingError}
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:77]: suggestion: This does more than reset the naming error
```suggestion
          onChange={onChangeHandler}
```

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+    if (isDuplicateName) {
+      setNamingError(isDuplicateName);
+    }
+  }, [isDuplicateName]);
+
+  // function to reset the name error when the name is changed
+  const onChangeResetNamingError = (e) => {
+    if (isNameChanged) {
+      setNamingError("");
+    }
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:61]:
```suggestion
  const onChangeHandler = (e) => {
    if (isNameChanged) {
      setNamingError("");
      setOptoutNamingError("");
    }
```

File: src/ListView/CreatePage/CreatePageModal.jsx [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+  function handleCreate() {
+    pagesApi
+      .createPage({ ...state, mode })
+      .then(function (response) {
+        if (response._data.editor_url) {
+          window.location.href = `${window.location.origin}${response._data.editor_url}`;
+        } else {
+          // this should never happen
+          setError("Something went wrong and we could not load the page.");
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageModal.jsx:24]: suggestion: `setError` should not use both booleans and strings
```suggestion
          setError(true);
```

File: src/ListView/CreatePage/CreatePageModal.jsx [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+        if (
+          serverErrorMessage &&
+          serverErrorMessage.includes(duplicateNameError)
+        ) {
+          setIsDuplicateName("Unique page name required.");
+        } else {
+          console.log("ERROR: ", serverError.error_message);
+          setError(
+            "An error occurred and the page could not be created. Please try again or contact support if the issue persists"
+          );
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageModal.jsx:42]:
```suggestion
          setError(true);
```

File: src/ListView/CreatePage/CreatePageModal.jsx [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+
+  return (
+    <Modal
+      title={title}
+      applyButtonText={getSubmitButtonText()}
+      disableApply={!state.name || !state.type}
+      onApply={handleCreate}
+      onDismiss={() => setDisplayModal(false)}
+    >
+      {error && <Notification type="error">{error}</Notification>}
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageModal.jsx:72]:
```suggestion
      {error && 
          <Notification type="error">
              An error occurred creating the page. Please try again or contact support if the issue persists.
          </Notification>}
```

**Review by lindsaycooper — CHANGES_REQUESTED** [review]

File: src/ListView/features/CreatePage/CreatePageForm.jsx (hunk 5 above)
Comment (shanthru) [reply src/ListView/features/CreatePage/CreatePageForm.jsx]: updated

File: src/ListView/features/CreatePage/CreatePageForm.jsx (hunk 6 above)
Comment (shanthru) [reply src/ListView/features/CreatePage/CreatePageForm.jsx]: updated

File: src/ListView/features/CreatePage/CreatePageForm.jsx (hunk 8 above)
Comment (shanthru) [reply src/ListView/features/CreatePage/CreatePageForm.jsx]: updated

File: src/ListView/features/CreatePage/CreatePageForm.jsx (hunk 11 above)
Comment (shanthru) [reply src/ListView/features/CreatePage/CreatePageForm.jsx]: changed to checkForNamingError

File: src/ListView/features/CreatePage/CreatePageModal.jsx (hunk 12 above)
Comment (shanthru) [reply src/ListView/features/CreatePage/CreatePageModal.jsx]: updated

File: src/ListView/features/CreatePage/CreatePageModal.jsx (hunk 13 above)
Comment (shanthru) [reply src/ListView/features/CreatePage/CreatePageModal.jsx]: removed this variable

File: src/ListView/features/CreatePage/CreatePageForm.jsx (hunk 16 above)
Comment (shanthru) [reply src/ListView/features/CreatePage/CreatePageForm.jsx]: updated

File: src/ListView/CreatePage/CreatePageForm.jsx (hunk 17 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx]: updated

File: src/ListView/CreatePage/CreatePageForm.jsx (hunk 18 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx]: updated

File: src/ListView/CreatePage/CreatePageForm.jsx (hunk 19 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx]: updated

File: src/ListView/CreatePage/CreatePageForm.jsx (hunk 20 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx]: updated

File: src/ListView/CreatePage/CreatePageModal.jsx (hunk 21 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageModal.jsx]: updated everything to string

File: src/ListView/CreatePage/CreatePageModal.jsx (hunk 22 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageModal.jsx:46]: updated everything to string

File: src/ListView/CreatePage/CreatePageModal.jsx (hunk 23 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageModal.jsx:76]: updated everything to string

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 24]
```diff
@@ -0,0 +1,126 @@
...
+  name = "",
+  onChange,
+  isDuplicateName,
+  isNameChanged,
+}) {
+  const [namingError, setNamingError] = useState("");
+  const [optoutNamingError, setOptoutNamingError] = useState("");
+
+  // function to check for empty, alphanumeric, and 'oc' page names
+  const checkForNamingError = function () {
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:21]:
```suggestion
  const checkForNamingError = function () {
```

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+    if (!nameWithoutDashes.match(alphanumericRegex)) {
+      setNamingError(
+        "The page name must only contain alphanumeric characters and dashes. Please use a different page name."
+      );
+      return;
+    }
+  };
+
+  // sets the error if the name is a duplicate (returned from the backend)
+  useEffect(() => {
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:51]:
```suggestion
  useEffect(() => {
```

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 26]
```diff
@@ -0,0 +1,126 @@
...
+    if (isDuplicateName) {
+      setNamingError(
+        "The page name must be unique. Please use a different page name."
+      );
+    }
+  }, [isDuplicateName]);
+
+  // function to reset the name error when the name is changed
+  const onChangeHandler = (e) => {
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:60]:
```suggestion
  const onChangeHandler = (e) => {
```

File: src/ListView/CreatePage/CreatePageModal.jsx [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+        if (response._data.editor_url) {
+          window.location.href = `${window.location.origin}${response._data.editor_url}`;
+        } else {
+          // this should never happen
+          setError(
+            // TODO move displaying this error to ListView (EPT-1713)
+            `Something went wrong and we could not load the new page: ${response._data.name}`
+          );
+          console.error(
+            `Error redirecting to page name:${JSON.stringify(response._data)}`
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageModal.jsx:29]:
```suggestion
            `Error redirecting to page:${JSON.stringify(response._data)}`
```

File: src/ListView/CreatePage/CreatePageModal.jsx [hunk 28]
```diff
@@ -0,0 +1,87 @@
...
+        let serverErrorMessage = serverError.error_message;
+        if (
+          serverErrorMessage &&
+          serverErrorMessage.includes("There is already a page named")
+        ) {
+          setIsDuplicateName(true);
+        } else {
+          console.log("ERROR: ", serverError.error_message);
+          setError(
+            "An error occurred and the page could not be created. Please try again or contact support if the issue persists"
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageModal.jsx:45]:
```suggestion
            "An error occurred and the page could not be created. Please try again or contact support if the issue persists."
```

**Review by lindsaycooper — COMMENTED** [review]
A few nit-picks

**Review by lindsaycooper — APPROVED** [review]
LGTM

**Review by lindsaycooper — COMMENTED** [review]
### QA Steps

✅ Verify New Page button opens HTML modal to create a new page image
- ✅ Title of modal should read “New HTML Page”
- ✅ Page name explanation copy: `Page name must follow URL standards and only contain alphanumeric characters and dashes. It is recommended to use dashes instead of spaces.`
- ✅ Name field: Auto-focused on modal load
- ✅ Name field: Required for “Create” button to be clickable
- ✅ Name field: Verify “Create” button is **enabled** when name has 1 alphanumeric character (when Category selected)
- Name field: Verify “Create” button is disabled when name doesn’t have an alphanumeric character (when Category selected). Error message: `The page name must only contain alphanumeric characters and dashes. Please use a different page name.`
  - ✅ When the "Create" button has **not** previously been enabled the above is true
  - ✅ When the "Create" button has been enabled and then the name is removed, the above is true
  - ❌ When the "Create" button has been enabled and then a new invalid name is entered, the "Create" button is not disabled on error 
  image

- ✅ Name field: Verify removing text from this field disables the button (when Category selected)
- ✅ Name field: Must be unique
- ✅ Name field:  Show error message under name when backend responds with 4xx error. Error Message: `The page name must be unique. Please use a different page name.`
- ✅ Category field: No buttons are selected on modal load
- ✅ Category field: Required for “Create” button to be clickable
- ✅ Category field: Verify “Create” button is **disabled** when no radio button selected (when Name field is valid)
- ✅ Category field: Verify “Create” button is **enabled** when radio button selected (when Name field is valid)

- ✅ Verify that when selecting “CANCEL” in the modal, modal is removed and returned back to List Page
- ✅ Verify that when selecting “X” in the modal, modal is removed and returned back to List Page

- ✅ Verify that when selecting “CREATE”  in the modal with valid name and category, redirected to HTML editor

@shanthru I think this error message should be updated to this: 
```The page name must contain at least one alphanumeric character and may contain dashes. Please use a different page name.```
 image

File: src/ListView/CreatePage/CreatePageForm.jsx (hunk 24 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx]: function comments have been removed

File: src/ListView/CreatePage/CreatePageForm.jsx (hunk 25 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx]: function comments have been removed

File: src/ListView/CreatePage/CreatePageForm.jsx (hunk 26 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx]: function comments have been removed

File: src/ListView/CreatePage/CreatePageModal.jsx (hunk 28 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageModal.jsx]: updated

**Comment by shanthru** [comment]
A change has been made to handle the failing test case.
I also noted one other UI display issue and implemented a change for that in a separate commit.

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 29]
```diff
@@ -0,0 +1,131 @@
...
+  onChange,
+  isDuplicateName,
+  isNameChanged,
+  setIsValidName,
+}) {
+  const [namingError, setNamingError] = useState("");
+  const [optoutNamingError, setOptoutNamingError] = useState("");
+
+  const checkForNamingError = function () {
+    setNamingError("");
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:22]: question: Does `setIsValidName` need to be reset here too?
```suggestion
    setNamingError("");
    setIsValidName(true);
```

Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx:22]: yes, it should!

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 30]
```diff
@@ -0,0 +1,132 @@
...
+    setOptoutNamingError("");
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:23]: question: Are these needed here since you are resetting them in `onChangeHandler`?

File: src/ListView/CreatePage/CreatePageForm.jsx [hunk 31]
```diff
@@ -0,0 +1,132 @@
...
+      setOptoutNamingError("");
+      setIsValidName(true);
+    }
+    onChange({ name: e.target.value });
+  };
+
+  const onOptoutChangeHandler = (e) => {
+    if (name === "optout") {
+      setOptoutNamingError(" "); //this must be a space and not an empty string or the error will clear
+    }
```
Comment (lindsaycooper) [inline src/ListView/CreatePage/CreatePageForm.jsx:73]: I'm confused by this use case. Can you explain more why the space is needed? Why not put an actual error message? Does this negation do the same thing?
```suggestion
    if (name !== "optout") {
      setOptoutNamingError("");
    }
```

Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx]: yes it does!

File: package.json [hunk 32]
```diff
@@ -22,6 +22,7 @@
   "dependencies": {
     "@sailthru/stui-components": "2.25.1",
     "@sailthru/stui-elements": "2.25.1",
+    "@sailthru/stui-layouts": "1.0.5",
```
[the remark this answers is not in the document]
Comment (shanthru) [reply package.json:25]: `stui-layouts` [latest](https://github.com/sailthru/stui-layouts/releases/tag/v1.0.5) version is 1.0.5
I had tried another version of `design-tokens` but had conflicts until I switched to this version.

File: src/ListView/CreatePage/CreatePageForm.jsx (hunk 30 above)
Comment (shanthru) [reply src/ListView/CreatePage/CreatePageForm.jsx:23]: good point...
I originally had these set here before implementing the `onChangeHandler` (some of our UIs do not reset the error until the next onBlur)
There isn't a negative impact on removing them but checking to see if there is any best practice to leave this.

Comment (lindsaycooper) [reply src/ListView/CreatePage/CreatePageForm.jsx:23]: I think this would fall under DRY principle - if it's only needed one place, only put it in one place

**Review by lindsaycooper — APPROVED** [review]
CR +1

**Review by lindsaycooper — APPROVED** [review]
#### QA +1

✅  Verify New Page button opens HTML modal to create a new page
- ✅ Title of modal should read “New HTML Page”
- ✅ Page name explanation copy: `Page name must follow URL standards and only contain alphanumeric characters and dashes. It is recommended to use dashes instead of spaces.`
- ✅ Name field: Auto-focused on modal load
- ✅ Name field: Required for “Create” button to be clickable
- ✅ Name field: Verify “Create” button is **enabled** when name has 1 alphanumeric character (when Category selected)
- ✅ Name field: Verify “Create” button is disabled when name doesn’t have an alphanumeric character (when Category selected). Error message: `The page name must contain at least one alphanumeric character and may contain dashes. Please use a different page name.`
  - ✅ When the "Create" button has **not** previously been enabled
  - ✅ When the "Create" button has been enabled and then the name is removed
  - ✅ When the "Create" button has been enabled and then a new invalid name is entered
- ✅ Name field: Verify removing text from this field disables the button (when Category selected)
- ✅ Name field: Must be unique. Show error message under name when existing page already loaded in the list. Error Message: `The page name must be unique. Please use a different page name.`
- ✅ Name field:  Show error message under name when backend responds with 4xx error. Error Message: `The page name must be unique. Please use a different page name.`

- ✅ Category field: No buttons are selected on modal load
- ✅ Category field: Required for “Create” button to be clickable
- ✅ Category field: Verify “Create” button is **disabled** when no radio button selected (when Name field is valid)
- ✅ Category field: Verify “Create” button is **enabled** when radio button selected (when Name field is valid)

- ✅ Verify that when selecting “CANCEL” in the modal, modal is removed and returned back to List Page
  **❓When there is an error (including empty name), the user has to click the cancel button twice. This behavior also happens in the modals on the Custom Fonts page so I'm ok with this not being a blocker**
- ✅ Verify that when selecting “X” in the modal, modal is removed and returned back to List Page

- ✅ Verify that when selecting “CREATE”  in the modal with valid name and category, redirected to HTML editor for page with that name & category
  - ✅ Verify that "Signup" category selected creates a page with type "signup"
  - ✅ Verify that "User Management" category selected creates a page with type "manage"
  - ✅ Verify that "Other" category selected creates a page with type "other"

- ✅ Edge Case with name "optout"
  1. Type "Test" in name field
  2. Select "other" or "signup" in the category field. Side effect: "Create" button enabled
  3. Type "optout" in name field
    - ✅ Subtext displays: `The page name 'optout' requires 'User Management' as the category. Please select 'User Management' below.` 
    - ✅ Radio buttons  "Signup" and "Other" are disabled
    - ✅ "Create" button is disabled
